rp2040-hal/src/multicore.rs

```diff
@@ -0,0 +1,211 @@
+//! Multicore support
+// See [Chapter ?? Section ??](https://datasheets.raspberrypi.org/rp2040/rp2040_datasheet.pdf) for more details
+
+use crate::pac;
+
+#[cfg(feature = "alloc")]
+extern crate alloc;
+
+/// Errors for multicore operations.
+#[derive(Debug)]
+pub enum Error {
+    /// Operation is invalid on this core.
+    InvalidCore,
+    /// Core was unresposive to commands.
+    Unresponsive,
+}
+
+// We pass data to cores via the stack, so we read
+// the data off the stack and into parameters that
+// rust can read here. Ideally this would be a
+// #[naked] function but that is not stable yet.
+static MULTICORE_TRAMPOLINE: [u16; 2] = [
+    0xbd03, // pop {r0, r1, pc} - call wrapper (pc) with r0 and r1
+    0x46c0, // nop - pad this out to 32 bits long
+];
+static mut CORE1_STACK: [usize; 4096] = [0; 4096];
```

Comment:
We should make an issue about making this a feature to allow the user to provide their own stack.
Also, need to remember to align this so we aren't wasting 24bytes of memory.